job_controller/status_test.go

```diff
@@ -0,0 +1,66 @@
+package job_controller
+
+import (
+	"testing"
+
+	"github.com/kubeflow/common/operator/v1"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func TestIsSucceeded(t *testing.T) {
+	jobStatus := v1.JobStatus{
+		Conditions: []v1.JobCondition{
+			{
+				Type:   v1.JobSucceeded,
+				Status: corev1.ConditionTrue,
+			},
+		},
+	}
+	assert.True(t, isSucceeded(jobStatus))
+}
+func TestUpdateJobConditions(t *testing.T) {
+	jobStatus := v1.JobStatus{
+
+	}
+	conditionType := v1.JobCreated
+	reason := "Job Created"
+	message := "Job Created"
+
+	err := updateJobConditions(&jobStatus, conditionType, reason, message)
+	if assert.NoError(t, err) {
+		conditionInStatus := jobStatus.Conditions[0]
+		assert.Equal(t, conditionInStatus.Type, conditionType)
+		assert.Equal(t, conditionInStatus.Reason, reason)
+		assert.Equal(t, conditionInStatus.Message, message)
+	}
+}
+
+func TestUpdateJobReplicaStatuses(t *testing.T) {
+	jobStatus := v1.JobStatus{}
+	initializeReplicaStatuses(&jobStatus, "worker")
```

Comment:
Add asserts here for the initialized statuses

Reply:
added